db/system_auth_keyspace.cc

```diff
@@ -0,0 +1,49 @@
+/*
+ * Modified by ScyllaDB
+ * Copyright (C) 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: (AGPL-3.0-or-later and Apache-2.0)
+ */
+
+#include "system_auth_keyspace.hh"
+#include "system_keyspace.hh"
+#include "db/schema_tables.hh"
+#include "schema/schema_builder.hh"
+
+namespace db::system_auth_keyspace {
+
+// use the same gc setting as system_schema tables
+using days = std::chrono::duration<int, std::ratio<24 * 3600>>;
+static constexpr auto auth_gc_grace = std::chrono::duration_cast<std::chrono::seconds>(days(7)).count();
+
+schema_ptr roles() {
```

Comment:
Are there changes I will not able to make even if tables have no data? Schema is hard-coded so I don't need to perform any migrations.

Reply:
I'm not entirely sure, but I wouldn't risk it. We already hide some tables like `system.topology` behind the consistent-topology-changes experimental flag, it's not very difficult to do so (see [here](https://github.com/scylladb/scylladb/blob/35da9551fbaa80c9c0603d5824d554c856fc8e1b/db/system_keyspace.cc#L2093-L2095)).

Reply:
yes it's doable, if I recall there is more places I need to guard, code becomes uglier and I didn't see the risk in showing empty extra tables, but if you say there is any risk I can add feature guards

Reply:
Now I recall why I didn't do it. It's about `static const std::unordered_set<std::string_view> system_keyspaces` which is static so can't be modified conditionally. It's mainly used by `bool is_system_keyspace` which is called from a few places and finding config reference in all of them may be problematic.

Reply:
> Now I recall why I didn't do it. It's about `static const std::unordered_set<std::string_view> system_keyspaces` which is static so can't be modified conditionally.

Right, I see that it's used by `is_system_keyspace` which, in turn is used from lots of different places...

I still think it's not worth risking it. I also understand that propagating `db::config` to all those places would be painful (and potentially wasteful because it will no longer be necessary as we move this out of experimental).

Maybe we could make `system_keyspaces` non-const, and conditionally append the auth_v2 keyspace name to the list right after the config is parsed? I admit it's a hack and I don't like it at all, but we will be able to easily and immediately remove it when moving the feature out of experimental state.

Reply:
will try to craft the hack

Reply:
Would it be possible to add to `system_keyspaces` unconditionally but only create the keyspace/table conditionally?

Reply:
Adding the name only to `system_keyspaces` should not break anything even if the keyspace doesn't exist

Reply:
> Would it be possible to add to system_keyspaces unconditionally but only create the keyspace/table conditionally?

it's a hack too but can work because the only place it iterates over happily ignores missing keyspace. 

Added in https://github.com/scylladb/scylladb/pull/16578/commits/3ad6b5be952a62463dac7b9525674ea9efd29bba